src/main/java/net/snowflake/client/jdbc/cloud/storage/StorageClientFactory.java

```diff
@@ -63,6 +64,7 @@ public SnowflakeStorageClient createClient(
             stage.getCredentials(),
             parallel,
             encMat,
+            stage.getProxyProperties(),
```

Comment:
Do we really need to modify StageInfo class? Can SFSession help us to get enough info?

Reply:
I believe we do, because sometimes we upload to S3 and Azure via connectionless upload (no GS instance). With connectionless upload, SFSession is null so there needs to be some other way to transfer the proxy properties.

Reply:
Sounds good!